### PR TITLE
added push permissions for Android & iOS

### PR DESCRIPTION
**Pull request by thisisthekap**
### Description of Change ###

Added push notification permissions for Android and iOS to the `Permissons` component.

### Bugs Fixed ###

- Related to issue #2062

### API Changes ###

List all API changes here (or just put None), example:

Added: 
 
- Permission `Permissions.PushNotification`
- Permission `Permissions.ProvisionalPushNotification`

Changed:

None
 
### Behavioral Changes ###

None

### PR Checklist ###

- [ ] Has tests
- [ ] Has samples
- [x] Rebased on top of `main` at time of PR
- [x] Changes adhere to coding standard
- [ ] Updated documentation ([see walkthrough](https://github.com/xamarin/Essentials/wiki/Documenting-your-code-with-mdoc))
  - Omitted, because msbuild tasks given in link is failing

### CODE REVIEW

**Comment by thisisthekap** [comment]
@microsoft-github-policy-service agree [company="fun.music IT GmbH"]

**Comment by thisisthekap** [comment]
@microsoft-github-policy-service agree company="fun.music IT GmbH"

**Comment by trampster** [comment]
We also need this change

File: Xamarin.Essentials/Permissions/Permissions.ios.tvos.watchos.cs [hunk 1]
```diff
@@ -269,5 +270,74 @@ public partial class StorageWrite : BasePlatformPermission
...
+                        message: null),
+                };
+            }
+        }
+
+        public partial class PushNotification : BasePlatformPermission
+        {
+            public override Task<PermissionStatus> RequestAsync()
+            {
+                EnsureDeclared();
```
Comment (jfversluis) [inline Xamarin.Essentials/Permissions/Permissions.ios.tvos.watchos.cs:310]: There is no `RequiredInfoPlistKeys` override. I assume that we want to check if there is a plist entry/entries for push notifications?

**Review by jfversluis — CHANGES_REQUESTED** [review]

**Comment by thisisthekap** [comment]
Good idea. Will do.

**Comment by jamesmontemagno** [comment]
I am not sure how many large changes we want to introduce into Xamarin.Essentials where new work should be done directly in .NET MAUI. I only recommend small bug fixes and maintenance IMHO.

This functionality could be achieved today via https://learn.microsoft.com/en-us/xamarin/essentials/permissions?context=xamarin%2Fandroid&tabs=android#extending-permissions

**Comment by thisisthekap** [comment]
@jfversluis As Xamarin.Forms is going to support Android 16 (a prerelease doing so is available at https://github.com/xamarin/Xamarin.Forms/releases/tag/beta-5.0.0-sr15-pre1), I think this PR should be reevaluated.

**Comment by lassana** [comment]
This has to be reevaluated. Xamarin (and Xamarin.Forms) is claimed to be supported till April 2024, and notification permission is a core part of Android 13 which Xamarin claims to support.

**Comment by Jerome-Liger** [comment]
Google force to update all our apps to target Android13 before 31/08/2023 on the PlayStore, so please, do not forget Xamarin.Forms apps IN PRODUCTION :  August 2023 -> April 2024 : it has to be implemented for me.
MAUI (and all the ecosystem) is far away to be ready to migrate all our xamarin.forms production apps...

**Comment by lndalmd** [comment]
Same here!!!!We are also concerned about Xamarin.Forms ending support until all of our project dependencies have been migrated to MAUI.Leandro AlmeidaEm 22 de ago. de 2023, à(s) 05:02, Jérôme Liger ***@***.***> escreveu:﻿
Google force to update all our apps to target Android13 before 31/08/2023 on the PlayStore, so please, do not forget Xamarin.Forms apps IN PRODUCTION :  August 2023 -> April 2024 : it has to be implemented for me.
MAUI (and all the ecosystem) is far away to be ready to migrate all our xamarin.forms production apps...

**Pull request closed without merging**